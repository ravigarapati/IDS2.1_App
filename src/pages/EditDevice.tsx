import React, {useEffect, useState} from 'react';
import {View, StyleSheet, Image, TouchableOpacity} from 'react-native';
import {BoschIcon, Button, CustomInputText} from '../components';
import {Dictionary} from '../utils/dictionary';
import {connect} from 'react-redux';
import {
  editDevice,
  getDeviceList2,
  getDeviceListEditDevice,
  updateDeviceList,
} from '../store/actions/HomeOwnerActions';
import {showToast} from '../components/CustomToast';
import {ScrollView} from 'react-native-gesture-handler';
import {Icons} from '../utils/icons';
import {Text} from 'react-native';

function EditDevice(props) {
  const [newDescriptionValue, setNewDescriptionValue] = useState(
    props.navigation.getParam('description'),
  );
  const [isChanged, setIsChanged] = useState(true);
  const isThermostat = props.navigation.getParam('isThermostat');

  useEffect(() => {
    let newDescription = props.navigation.getParam('description');
    let pos = 0;
    let result = [];
    if (newDescription.includes(' Thermostat')) {
      while (newDescription.indexOf(' Thermostat', pos) != -1) {
        result.push(newDescription.indexOf(' Thermostat', pos));
        pos = newDescription.indexOf(' Thermostat', pos) + 1;
      }
      setNewDescriptionValue(
        newDescription.substring(0, result[result.length - 1]),
      );
    }
  }, []);

  const save = () => {
    let deviceType = '';
    props.navigation.state.params.createStatusInterval();
    /*setTimeout(() => {
      props.navigation.state.params.createStatusInterval();
    }, 3000);*/
    if (props.navigation.getParam('deviceType').includes('IDS')) {
      if (props.navigation.getParam('deviceType') === 'IDS Premium Connected') {
        deviceType = 'IDS2.1';
      } else {
        deviceType = 'IDS3.0';
      }
    } else {
      deviceType = props.navigation.getParam('deviceType');
    }
    props.editDevice({
      deviceName: newDescriptionValue,
      deviceId: props.navigation.getParam('macId'),
      isThermostat: isThermostat,
      //idsBody: {
      //  gatewayId: props.navigation.getParam('macId'),
      //  ODUName: newDescriptionValue,
      //  ODUInstalledAddress: props.navigation.getParam('installationAddress'),
      //  contractorMonitoringStatus: props.navigation.getParam(
      //    'contractorMonitoringStatus',
      //  ),
      //  addressChanged: false,
      //},
      deviceType: deviceType,
    });
    props.navigation.navigate('HomeTabs');
  };

  const cancel = () => {
    props.navigation.state.params.createStatusInterval();

    props.navigation.navigate('HomeTabs');
  };

  const onBack = () => {
    cancel();
  };

  return (
    <View style={{flex: 1}}>
      <>
        <View style={styles.headerContainer}>
          <View style={styles.headerDivision}>
            <TouchableOpacity
              testID="backButton"
              style={styles.headerBackButton}
              onPress={() => onBack()}>
              <BoschIcon
                name={Icons.backLeft}
                size={24}
                style={styles.marginHorizontal10}
              />
            </TouchableOpacity>
          </View>

          <View style={styles.headerTitle}>
            <Text
              style={{
                fontSize: 21,
                marginVertical: 10,
              }}>
              {'Edit Device'}
            </Text>
          </View>
        </View>
        <Image
          style={styles.headerRibbon}
          source={require('../assets/images/header_ribbon.png')}
        />
      </>
      <ScrollView
        keyboardShouldPersistTaps="handled"
        style={{flex: 1}}
        contentContainerStyle={{flexGrow: 1}}>
        <View style={styles.container}>
          <View>
            <CustomInputText
              allowFontScaling={true}
              testID="newName"
              value={newDescriptionValue}
              placeholder={
                isThermostat
                  ? Dictionary.addDevice.addBcc.deviceLocation
                  : Dictionary.myAppliance.unitName
              }
              maxLength={15}
              onChange={text => {
                setNewDescriptionValue(text.replace(/[^a-zA-Z 0-9]/g, ''));
                setIsChanged(false);
              }}
              tooltip={
                isThermostat ? Dictionary.addDevice.addBcc.tooltipMessage : ''
              }
            />
            <CustomInputText
              allowFontScaling={true}
              value={props.navigation.getParam('macId')}
              placeholder={
                isThermostat
                  ? Dictionary.addDevice.addBcc.macId
                  : Dictionary.addDevice.addBcc.serialNumber
              }
              disabled={true}
            />
          </View>
          <View>
            <Button
              testID="submit"
              disabled={newDescriptionValue === '' || isChanged}
              text={Dictionary.button.save}
              onPress={save}
              type="primary"
            />
            <Button
              testID="cancel"
              text={Dictionary.button.cancel}
              onPress={cancel}
              type="secondary"
            />
          </View>
        </View>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    paddingHorizontal: 15,
    paddingTop: 20,
    justifyContent: 'space-between',
  },
  headerContainer: {
    backgroundColor: '#ffff',
    flexDirection: 'row',
    alignItems: 'center',
  },
  headerDivision: {
    flexDirection: 'column',
    flex: 0.15,
    backgroundColor: '#ffff',
    justifyContent: 'center',
  },
  headerBackButton: {
    justifyContent: 'center',
    flex: 1,
  },
  headerTitle: {
    flexDirection: 'column',
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'center',
    marginLeft: 24,
  },
  headerRibbon: {height: 8, width: '100%'},
  confirmationPageContainer: {
    flex: 1,
    flexDirection: 'column',
    paddingHorizontal: 20,
    backgroundColor: 'white',
  },
  marginHorizontal10: {marginHorizontal: 10},
});

const mapDispatchToProps = {
  editDevice,
  getDeviceList2,
  getDeviceListEditDevice,
  updateDeviceList,
};

export default connect(null, mapDispatchToProps)(EditDevice);
