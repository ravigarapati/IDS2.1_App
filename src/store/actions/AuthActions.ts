import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  CURRENT_USER,
  CURRENT_STEP,
  UPDATE_CURRENT_USER,
  LOGOUT,
  TERMS_AND_CONDITIONS,
  LATEST_VERSION,
  SET_STEP_TITLES,
  SET_USER_ROLE,
  SET_SHOW_ROLE,
  SET_USER_ADDRESS,
  RESET_USER,
  SET_USER_NAME,
} from '../labels/AuthLabels';
import {Auth} from 'aws-amplify';
import {showToast} from '../../components/CustomToast';
import {
  apiInterceptorForPost,
  apiInterceptorForPostCustomApi,
  openApiInterceptorForGet,
  apiInterceptorForPostAPI,
} from './CommonActions';
import ClearCache from 'oa-react-native-clear-cache';

export const setCurrentUser = user => {
  return async dispatch => {
    dispatch({type: CURRENT_USER, user: user});
    ClearCache.clearAppCache(() => {});
  };
};
export const setCurrentStep = stepNumber => {
  return async dispatch => {
    dispatch({type: CURRENT_STEP, data: stepNumber});
  };
};
export const updateUserAttributes = (userFromStore, data, callback?) => {
  return async dispatch => {
    Auth.updateUserAttributes(userFromStore, data)
      .then(() => {
        if (callback) {
          callback();
        }
      })
      .catch(error => {
        showToast(error.message, 'error');
      });
  };
};

export const verifyCurrentUserAttribute = attribute => {
  Auth.verifyCurrentUserAttribute(attribute)
    .then(() => {})
    .catch(error => {
      showToast(error.message, 'error');
    });
};

export const verifyCurrentUserAttributeSubmit = (
  attribute,
  value,
  callback?,
) => {
  Auth.verifyCurrentUserAttributeSubmit(attribute, value)
    .then(() => {
      if (callback) {
        callback();
      }
    })
    .catch(error => {
      showToast(error.message, 'error');
    });
};

export const updateUserObject = attributes => {
  return async dispatch => {
    dispatch({type: UPDATE_CURRENT_USER, data: attributes});
  };
};

export const checkAdminDetails = (attr, callback?) => {
  return async dispatch => {
    const path = '/checkadmindetails';
    apiInterceptorForPost(
      dispatch,
      path,
      attr,
      response => {
        callback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const createProfile = (userData, callback?) => {
  return async dispatch => {
    const path = '/registration';
    apiInterceptorForPost(
      dispatch,
      path,
      userData,
      response => {
        callback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const createProfileHomeOwner = (userData, callback?) => {
  return async dispatch => {
    const baseUrl =
      'https://r9nzt6y3ui.execute-api.us-east-1.amazonaws.com/development';
    apiInterceptorForPostCustomApi(
      dispatch,
      baseUrl,
      '/user/register',
      {
        firstName: userData.firstName,
        lastName: userData.lastName,
        phoneNumber: userData.phoneNumber,
        type: 'HomeOwner',
        timestamp: `${new Date().valueOf()}`,
      },
      response => {
        if (response.message) {
          if (
            response.message === 'Operation succeed' ||
            response.message.startsWith("'e")
          ) {
            callback();
          } else if (
            response.message === 'The user account has already registered'
          ) {
            showToast(response.message, 'info');
          } else {
            showToast(response.message, 'error');
          }
        }
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const logout = () => {
  return async dispatch => {
    dispatch({type: LOGOUT, data: {}});
    // clear the storage
    ClearCache.getAppCacheSize(data => {});
    ClearCache.clearAppCache(() => {});
    await AsyncStorage.removeItem('persist:root').then(() => {});
  };
};

export const getTermsandcondition = () => {
  return async dispatch => {
    const path = '/termsandconditions';
    openApiInterceptorForGet(
      dispatch,
      path,
      response => {
        dispatch({type: TERMS_AND_CONDITIONS, data: response.data});
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const updateTermsAndConditions = (input, successCallback) => {
  return async dispatch => {
    const path = '/termsandconditions';
    apiInterceptorForPost(
      dispatch,
      path,
      input,
      response => {
        successCallback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const deleteUser = (userRole, successCallback) => {
  return async dispatch => {
    const path = `/deleteuser/${userRole.type}`;
    apiInterceptorForPost(
      dispatch,
      path,
      {},
      response => {
        successCallback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const getLatestVersion = (successCallback?: any) => {
  return async dispatch => {
    const path = '/latestappversion';
    openApiInterceptorForGet(
      dispatch,
      path,
      response => {
        dispatch({type: LATEST_VERSION, data: response.data});
        successCallback && successCallback(response.data);
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const setStepTitles = stepTitles => {
  return async dispatch => {
    dispatch({type: SET_STEP_TITLES, data: stepTitles});
  };
};

export const setUserRole = userRole => {
  return async dispatch => {
    dispatch({type: SET_USER_ROLE, data: userRole});
  };
};

export const setShowRole = showRole => {
  return async dispatch => {
    dispatch({type: SET_SHOW_ROLE, data: showRole});
  };
};

export const setUserName = userName => {
  return async dispatch => {
    dispatch({type: SET_USER_NAME, data: userName});
  };
};

export const setUserAddress = userAddress => {
  return async dispatch => {
    dispatch({type: SET_USER_ADDRESS, data: userAddress});
  };
};

export const resetUser = () => {
  return async dispatch => {
    dispatch({type: RESET_USER});
  };
};