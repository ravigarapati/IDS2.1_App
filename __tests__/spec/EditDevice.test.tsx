import 'react-native';
import React from 'react';
import AddBcc from './../../src/pages/AddBcc';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import EditDevice from '../../src/pages/EditDevice';
import {render, cleanup, fireEvent} from 'react-native-testing-library';
import {createStore, combineReducers, applyMiddleware} from 'redux';
import HomeOwnerReducer from './../../src/store/reducers/HomeOwnerReducer';
import * as homeOwnerActions from './../../src/store/actions/HomeOwnerActions';
import thunk from 'redux-thunk';

jest.useFakeTimers();

const mockStore = configureMockStore();
const store = mockStore({
  homeOwner: {
    energyData: {},
    deviceList: [],
    selectedDevice: {},
    contractorMonitoringStatus: false,
    fanMode: '',
    fanOnFor: '',
    fanOffFor: '',
    fanIsScheduled: '',
    fanScheduledStart: '',
    fanScheduledEnd: '',
    selectedSchedule: '0',
    selectedScheduleName: '',
    scheduleInfo: {},
    prevSelectedDevice: {},
    deviceStatus: {},
    deviceInformation: {},
    deviceDetails: {},
    heatPumpInfo: {},
    reloadHeatPumpInfo: false,
    notificationList: [],
    lastKey: {},
    notificationsCount: 0,
    lastUpdated: null,
    isTermsConditionsAccepted: null,
    hoUserAnalytics: null,
    selectedUnitName: {},
    isThermostatSelected: true,
    isUserFirstLogin: true,
    actualWeatherOnFahrenheit: true,
    hapticVibration: true,
    _24HrsFormatSelected: false,
    location: 'Alabaster',
    updatedSelectedDevice: false,
    weatherInfo: {},
    weatherInfoLocation: {},
    locationSuggestions: [],
    locationInformation: {},
    idsSelectedDeviceAccess: '',
    utilyEnergySaving: '',
    idsSelectedDevice: '',
    idsSelectedOdu: '',
    idsSelectedDeviceType: '',
    tempLocation: '',
    newDeviceInfo: {},
    previousBcc: '',
    deviceList2: [
      /*{
        id: 2,
        //macId: 'e8fdf8a4773c',
        macId: '34eae7c351e6',
        deviceName: 'Bedroom Thermostat',
        deviceType: 'BCC100',
        isThermostat: true,
        isOn: true,
        mode: 0,
        setPoint: 74,
        heatingSetpoint: 74,
        coolingSetpoint: 79,
        current: 76,
        isAccesories: true,
        accesories: [
          {
            id: 0,
            name: 'Dehumidifier',
            isEnable: 0,
            isOn: false,
          },
          {
            id: 1,
            name: 'Humidifier',
            isEnable: 1,
            isOn: false,
          },
        ],
        isMonitoring: null,
        isOnSchedule: true,
        stage: 3,
        isConnected: true,
        acceleratedHeating: true,
        schedules: [],
      },*/
    ],
  },
  auth: {},
});
const rootReducer = combineReducers({
  homeOwner: HomeOwnerReducer,
});
const realStore = createStore(
  rootReducer,
  {
    homeOwner: {
      energyData: {},
      deviceList: [],
      selectedDevice: {},
      contractorMonitoringStatus: false,
      fanMode: '',
      fanOnFor: '',
      fanOffFor: '',
      fanIsScheduled: '',
      fanScheduledStart: '',
      fanScheduledEnd: '',
      selectedSchedule: '0',
      selectedScheduleName: '',
      scheduleInfo: {},
      prevSelectedDevice: {},
      deviceStatus: {},
      deviceInformation: {},
      deviceDetails: {},
      heatPumpInfo: {},
      reloadHeatPumpInfo: false,
      notificationList: [],
      lastKey: {},
      notificationsCount: 0,
      lastUpdated: null,
      isTermsConditionsAccepted: null,
      hoUserAnalytics: null,
      selectedUnitName: {},
      isThermostatSelected: true,
      isUserFirstLogin: true,
      actualWeatherOnFahrenheit: true,
      hapticVibration: true,
      _24HrsFormatSelected: false,
      location: 'Alabaster',
      updatedSelectedDevice: false,
      weatherInfo: {},
      weatherInfoLocation: {},
      locationSuggestions: [],
      locationInformation: {},
      idsSelectedDeviceAccess: '',
      utilyEnergySaving: '',
      idsSelectedDevice: '',
      idsSelectedOdu: '',
      idsSelectedDeviceType: '',
      tempLocation: '',
      newDeviceInfo: {},
      previousBcc: '',
      deviceList2: [],
    },
  },
  applyMiddleware(thunk),
);

describe('Edit Device screen', () => {
  test('Component to be defined', () => {
    const navState = {
      params: {
        macId: 'testing',
        installationAddress: {},
        description: 'testing Thermostat',
        deviceType: 'BCC',
        contractorMonitoringStatus: false,
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <EditDevice navigation={navigation} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const navState = {
      params: {
        macId: 'testing',
        description: 'testing Thermostat',
        deviceType: 'BCC',
        installationAddress: {},
        contractorMonitoringStatus: false,
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <EditDevice navigation={navigation} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Click on cancel button', () => {
    let currentScreen = '';
    const navState = {
      params: {
        macId: 'testing',
        description: 'testing Thermostat',
        deviceType: 'BCC',
        installationAddress: {},
        contractorMonitoringStatus: false,
        createStatusInterval: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      navigate: screen => {
        currentScreen = screen;
      },
      state: navState,
    };
    const rendered = render(
      <Provider store={store}>
        <EditDevice navigation={navigation} />
      </Provider>,
    );
    let cancelButton = rendered.getByTestId('cancel');
    fireEvent(cancelButton, 'press');

    expect(currentScreen).toBe('HomeTabs');
  });

  test('Edit the device', () => {
    let currentScreen = '';
    let mockedValues = {
      deviceId: '98d863d2416a',
      deviceName: 'ab Thermostat',
    };
    const navState = {
      params: {
        macId: 'testing',
        description: 'testing Thermostat',
        installationAddress: {},
        contractorMonitoringStatus: false,
        deviceType: 'BCC',
        createStatusInterval: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      navigate: screen => {
        currentScreen = screen;
      },
      state: navState,
    };
    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => ['new name', () => null])
      .mockImplementationOnce(() => [false, () => null]);
    const rendered = render(
      <Provider store={realStore}>
        <EditDevice navigation={navigation} />
      </Provider>,
    );

    let editSpy = jest
      .spyOn(homeOwnerActions, 'editDevice')
      .mockReturnValueOnce(mockedValues);

    let submitButton = rendered.getByTestId('submit');
    fireEvent(submitButton, 'press');
    let valuesFromApi = editSpy();
    expect(valuesFromApi.deviceName).toBe(mockedValues.deviceName);
  });

  test('Edit the device (IDS 2.1)', () => {
    let currentScreen = '';
    let mockedValues = {
      deviceId: '399A-123-123123-8733955691',
      deviceName: 'Heat Pump 2',
    };
    const navState = {
      params: {
        macId: '399A-123-123123-8733955691',
        description: 'Heat Pump 1',
        installationAddress: {},
        contractorMonitoringStatus: false,
        deviceType: 'IDS Premium Connected',
        createStatusInterval: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      navigate: screen => {
        currentScreen = screen;
      },
      state: navState,
    };
    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => ['new name', () => null])
      .mockImplementationOnce(() => [false, () => null]);
    const rendered = render(
      <Provider store={realStore}>
        <EditDevice navigation={navigation} />
      </Provider>,
    );

    let editSpy = jest
      .spyOn(homeOwnerActions, 'editDevice')
      .mockReturnValueOnce(mockedValues);

    let submitButton = rendered.getByTestId('submit');
    fireEvent(submitButton, 'press');
    let valuesFromApi = editSpy();
    expect(valuesFromApi.deviceName).toBe(mockedValues.deviceName);
  });

  test('Edit the device (IDS 3.0)', () => {
    let currentScreen = '';
    let mockedValues = {
      deviceId: '399A-123-123123-8733955691',
      deviceName: 'Heat Pump 2',
    };
    const navState = {
      params: {
        macId: '399A-123-123123-8733955691',
        description: 'Heat Pump 1',
        installationAddress: {},
        contractorMonitoringStatus: false,
        deviceType: 'IDS Arctic',
        createStatusInterval: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      navigate: screen => {
        currentScreen = screen;
      },
      state: navState,
    };
    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => ['new name', () => null])
      .mockImplementationOnce(() => [false, () => null]);
    const rendered = render(
      <Provider store={realStore}>
        <EditDevice navigation={navigation} />
      </Provider>,
    );

    let editSpy = jest
      .spyOn(homeOwnerActions, 'editDevice')
      .mockReturnValueOnce(mockedValues);

    let submitButton = rendered.getByTestId('submit');
    fireEvent(submitButton, 'press');
    let valuesFromApi = editSpy();
    expect(valuesFromApi.deviceName).toBe(mockedValues.deviceName);
  });

  test('Click on back button', () => {
    let currentScreen = '';
    const navState = {
      params: {
        macId: 'testing',
        description: 'testing Thermostat',
        deviceType: 'BCC',
        installationAddress: {},
        contractorMonitoringStatus: false,
        createStatusInterval: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      navigate: screen => {
        currentScreen = screen;
      },
      state: navState,
    };
    const rendered = render(
      <Provider store={store}>
        <EditDevice navigation={navigation} />
      </Provider>,
    );
    let cancelButton = rendered.getByTestId('backButton');
    fireEvent(cancelButton, 'press');

    expect(currentScreen).toBe('HomeTabs');
  });
});
