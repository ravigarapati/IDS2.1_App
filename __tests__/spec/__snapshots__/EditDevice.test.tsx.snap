// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Device screen Component to be rendered 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffff",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffff",
          "flex": 0.15,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="backButton"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 24,
              },
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "fontFamily": "Bosch-Icon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î‚ 
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 24,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 21,
            "marginVertical": 10,
          }
        }
      >
        Edit Device
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/header_ribbon.png",
      }
    }
    style={
      Object {
        "height": 8,
        "width": "100%",
      }
    }
  />
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    keyboardShouldPersistTaps="handled"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    waitFor={
      Array [
        Object {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "justifyContent": "space-between",
            "paddingHorizontal": 15,
            "paddingTop": 20,
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "paddingVertical": 10,
                },
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "BoschSans-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                    "paddingHorizontal": 10,
                  },
                  Object {},
                ]
              }
            >
              Unit Name
            </Text>
            <View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <TextInput
                  accessibilityHint=""
                  accessibilityLabel=""
                  autoCapitalize="none"
                  autoCorrect={true}
                  blurOnSubmit={true}
                  editable={true}
                  keyboardType="default"
                  maxLength={15}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholderTextColor="#8A9097"
                  secureTextEntry={false}
                  selectTextOnFocus={true}
                  spellCheck={true}
                  style={
                    Array [
                      undefined,
                      Object {
                        "borderBottomWidth": 1,
                        "color": "#000000",
                        "fontFamily": "BoschSans-Regular",
                        "fontSize": 16,
                        "lineHeight": 22.4,
                        "paddingHorizontal": 10,
                      },
                      Object {
                        "paddingVertical": 10,
                      },
                      Object {
                        "borderBottomColor": "#000000",
                      },
                      false,
                      Object {},
                      Object {
                        "width": "100%",
                      },
                      false,
                    ]
                  }
                  testID="newName"
                  underlineColorAndroid="transparent"
                  value="testing"
                />
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        undefined,
                        Object {
                          "borderBottomWidth": 1,
                          "paddingHorizontal": 5,
                        },
                        Object {
                          "paddingVertical": 10,
                        },
                        Object {
                          "borderBottomColor": "#000000",
                        },
                        Object {
                          "padding": 0,
                        },
                        false,
                      ],
                      Object {
                        "fontFamily": "Bosch-Icon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#8A9097",
                      "fontSize": 17,
                      "left": 10,
                      "position": "absolute",
                      "zIndex": -1,
                    },
                    Object {
                      "top": 10,
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "position": "absolute",
                  "right": 10,
                  "top": 10,
                }
              }
            >
              <Image />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "paddingVertical": 10,
                },
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "BoschSans-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                    "paddingHorizontal": 10,
                  },
                  Object {
                    "color": "rgba(191, 192, 194, 1)",
                  },
                ]
              }
            >
              Serial Number
            </Text>
            <View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <TextInput
                  accessibilityHint=""
                  accessibilityLabel=""
                  autoCapitalize="none"
                  autoCorrect={true}
                  blurOnSubmit={true}
                  editable={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholderTextColor="#8A9097"
                  secureTextEntry={false}
                  selectTextOnFocus={false}
                  spellCheck={true}
                  style={
                    Array [
                      undefined,
                      Object {
                        "borderBottomWidth": 1,
                        "color": "#000000",
                        "fontFamily": "BoschSans-Regular",
                        "fontSize": 16,
                        "lineHeight": 22.4,
                        "paddingHorizontal": 10,
                      },
                      Object {
                        "paddingVertical": 10,
                      },
                      Object {
                        "borderBottomColor": "#C1C7CC",
                      },
                      false,
                      Object {
                        "color": "rgba(191, 192, 194, 1)",
                      },
                      Object {
                        "width": "100%",
                      },
                      false,
                    ]
                  }
                  testID=""
                  underlineColorAndroid="transparent"
                  value="testing"
                />
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        undefined,
                        Object {
                          "borderBottomWidth": 1,
                          "paddingHorizontal": 5,
                        },
                        Object {
                          "paddingVertical": 10,
                        },
                        Object {
                          "borderBottomColor": "#C1C7CC",
                        },
                        Object {
                          "padding": 0,
                        },
                        false,
                      ],
                      Object {
                        "fontFamily": "Bosch-Icon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#8A9097",
                      "fontSize": 17,
                      "left": 10,
                      "position": "absolute",
                      "zIndex": -1,
                    },
                    Object {
                      "top": 10,
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "position": "absolute",
                  "right": 10,
                  "top": 10,
                }
              }
            >
              <Image />
            </View>
          </View>
        </View>
        <View>
          <View>
            <View
              accessibilityElementsHidden={true}
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={true}
              importantForAccessibility="no-hide-descendants"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "marginVertical": 5,
                    "paddingVertical": 15,
                  },
                  Object {},
                  Object {
                    "backgroundColor": "#004975",
                  },
                  Object {
                    "backgroundColor": "#C1C7CC",
                  },
                  Object {
                    "borderColor": "#C1C7CC",
                    "borderWidth": 1,
                  },
                ]
              }
              testID="submit"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  accessible={true}
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                      Object {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {
                        "fontFamily": "BoschSans-Regular",
                      },
                      Object {},
                    ]
                  }
                >
                  Save
                </Text>
              </View>
            </View>
          </View>
          <View>
            <View
              accessibilityElementsHidden={false}
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              importantForAccessibility="auto"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "marginVertical": 5,
                    "paddingVertical": 15,
                  },
                  Object {},
                  Object {},
                  Object {
                    "borderColor": "#004975",
                    "borderWidth": 1,
                  },
                ]
              }
              testID="cancel"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  accessible={true}
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#004975",
                      },
                      Object {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {
                        "fontFamily": "BoschSans-Regular",
                      },
                      Object {},
                    ]
                  }
                >
                  Cancel
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
